ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/OperationPool.java

```diff
@@ -31,16 +35,32 @@
 
 public class OperationPool<T extends SszData> {
   private static final int OPERATION_POOL_SIZE = 1000;
+  private static final String OPERATION_POOL_SIZE_METRIC = "eth1_operation_pool_count_for_";
+  private static final String OPERATION_POOL_SIZE_VALIDATION_REASON =
+      "eth1_operation_pool_validation_reason_for_";
   private final Set<T> operations = LimitedSet.create(OPERATION_POOL_SIZE);
   private final Function<UInt64, SszListSchema<T, ?>> slotToSszListSchemaSupplier;
   private final OperationValidator<T> operationValidator;
   private final Subscribers<OperationAddedSubscriber<T>> subscribers = Subscribers.create(true);
+  private final LabelledMetric<Counter> validationReasonCounter;
 
   public OperationPool(
+      String metricType,
+      MetricsSystem metricsSystem,
       Function<UInt64, SszListSchema<T, ?>> slotToSszListSchemaSupplier,
       OperationValidator<T> operationValidator) {
     this.slotToSszListSchemaSupplier = slotToSszListSchemaSupplier;
     this.operationValidator = operationValidator;
+    metricsSystem.createIntegerGauge(
+        TekuMetricCategory.BEACON,
+        OPERATION_POOL_SIZE_METRIC + metricType,
+        "Count of items present in each type of pool",
+        this::size);
+    validationReasonCounter =
+        metricsSystem.createLabelledCounter(
+            TekuMetricCategory.BEACON,
+            OPERATION_POOL_SIZE_VALIDATION_REASON + metricType,
+            "track validation reason");
```

Comment:
This may be causing issues - `track validation reason` won't be a valid metric label name because of the spaces.  I'd suggest just `result`.